crates/crypto/bls/src/lib.rs

```diff
@@ -0,0 +1,19 @@
+pub mod aggregate_pubkey;
+pub mod constants;
+pub mod errors;
+pub mod pubkey;
+pub mod signature;
+pub mod traits;
+
+pub use aggregate_pubkey::AggregatePubKey;
+pub use pubkey::PubKey;
+pub use signature::BLSSignature;
+
+// The BLS (Boneh-Lynn-Shacham) cryptographic backend implementation is determined
+// at compile time via feature flags. Two implementations are supported:
+// - "supranational": Uses the supranational/blst library, optimized for performance
+// - "zkcrypto": Uses the zkcrypto/bls12_381 library implementation, optimized for zkVMs
```

Comment:
```suggestion
//! The BLS (Boneh-Lynn-Shacham) cryptographic backend implementation is determined
//! at compile time via feature flags. Two implementations are supported:
//! - "supranational": Uses the supranational/blst library, optimized for performance
//! - "zkcrypto": Uses the zkcrypto/bls12_381 library implementation, optimized for zkVMs

pub mod aggregate_pubkey;
pub mod constants;
pub mod errors;
pub mod pubkey;
pub mod signature;
pub mod traits;

pub use aggregate_pubkey::AggregatePubKey;
pub use pubkey::PubKey;
pub use signature::BLSSignature;

```

Most this to the top of the file  and make it inner doc comment's. Could you also make a README for the crate explaining how we support 2 different bls libraries which ones and examples and how to use the crate

Reply:
All reviews are resolved at https://github.com/ReamLabs/ream/pull/117/commits/9fa566ba68ac3909ca6a5f8252570951f7ccaa83! Thanks 👍